### PR TITLE
added collation to sql database

### LINKED ISSUE #1031: Mysql & Sql-Server does a case insensitive search on column with type string
### Describe the bug
Assume table customers with schema
**type customers {
id : ID! @primary
name : String!
}**
Having 2 rows
**id                    name
1                     john
2                     John**

Executing the SQL query **Select * from customers where name = 'john'** through MySQL shell returns 2 rows, as by default MySQL does a case insensitive search on a column with type text or varchar.
Same applies for SQL-server.

But if we execute the following graphql query through space-cloud
**query {
customers(where : {name : john}) @mysql {
id 
name
}**
Returns only 1 row instead of 2, as space cloud does some post-processing on the result( in utils.Validate() function )

### Expected behaviour
Even though MySQL & SQL-server do case insensitive search. space cloud should force case sensitive search by adding **collate** field while creating tables in space cloud.

For case insensitive search, developers can use **regex**

### Your environment
- Space Cloud version:
- OS:
- The client API version (if applicable):
- Browser (if applicable):

>If this bug restricts your use of space-cloud, give it a thumbs up reaction, so that we can determine which bugs need to be fixed immediately.
>👍

### PR DESCRIPTION
**Pull request by prithvi081099**
Fixes #1031

### CODE REVIEW

**Comment by sharadregoti** [comment]
@prithvi081099 your tests are failing

**Comment by sharadregoti** [comment]
also add collattion at dabase level

File: gateway/modules/schema/helpers.go [hunk 1]
```diff
@@ -93,6 +93,9 @@ func (c *creationModule) addNotNull() string {
 	case utils.Postgres:
 		return "ALTER TABLE " + c.schemaModule.getTableName(dbType, c.logicalDBName, c.TableName) + " ALTER COLUMN " + c.ColumnName + " SET NOT NULL"
 	case utils.SQLServer:
+		if c.columnType == "varchar("+model.SQLTypeIDSize+")" || c.columnType == "varchar(max)" || c.columnType == "text" {
+			return "ALTER TABLE " + c.schemaModule.getTableName(dbType, c.logicalDBName, c.TableName) + " ALTER COLUMN " + c.ColumnName + " " + c.columnType + " collate Latin1_General_CS_AS NOT NULL"
+		}
```
Comment (sharadregoti) [inline gateway/modules/schema/helpers.go:98]: remove text type in if case, Do this for all

File: gateway/utils/constants.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ import (
 )
 
 // BuildVersion is the current version of Space Cloud
-const BuildVersion = "0.19.0"
+const BuildVersion = "0.18.0"
```
Comment (YourTechBud) [inline gateway/utils/constants.go:8]:
```suggestion
const BuildVersion = "0.18.1"
```

File: gateway/modules/schema/helpers.go [hunk 3]
```diff
@@ -93,6 +93,9 @@ func (c *creationModule) addNotNull() string {
 	case utils.Postgres:
 		return "ALTER TABLE " + c.schemaModule.getTableName(dbType, c.logicalDBName, c.TableName) + " ALTER COLUMN " + c.ColumnName + " SET NOT NULL"
 	case utils.SQLServer:
+		if c.columnType == "varchar("+model.SQLTypeIDSize+")" || c.columnType == "varchar(max)" {
+			return "ALTER TABLE " + c.schemaModule.getTableName(dbType, c.logicalDBName, c.TableName) + " ALTER COLUMN " + c.ColumnName + " " + c.columnType + " collate Latin1_General_CS_AS NOT NULL"
+		}
```
Comment (YourTechBud) [inline gateway/modules/schema/helpers.go:98]: Whats the point of this? Maybe you should just check if there is a `varchar` prefix?

Comment (sharadregoti) [reply gateway/modules/schema/helpers.go:98]: yep

Comment (YourTechBud) [reply gateway/modules/schema/helpers.go:98]: Okay. So @prithvi081099 you'll need to change this everywhere

Comment (prithvi081099) [reply gateway/modules/schema/helpers.go]: I have made the changes @YourTechBud and @sharadregoti

File: gateway/utils/constants.go [hunk 4]
```diff
@@ -5,7 +5,7 @@ import (
 )
 
 // BuildVersion is the current version of Space Cloud
-const BuildVersion = "0.18.0"
+const BuildVersion = "0.18.1"
```
Comment (sharadregoti) [inline gateway/utils/constants.go:8]:
```suggestion
const BuildVersion = "0.18.0"
```

**Pull request merged**